packages/react/src/components/sideNavigation/SideNavigation.stories.tsx

```diff
@@ -0,0 +1,272 @@
+          href={ID.SUB_LEVEL_4}
+          label={labels.subLevel4}
+          onClick={handleClick(setActive)}
+        />
+      </SideNavigation.MainLevel>
+      <SideNavigation.MainLevel
+        active={ID.MAIN_LEVEL_3 === active}
+        id={ID.MAIN_LEVEL_3}
+        href={ID.MAIN_LEVEL_3}
+        label={labels.mainLevel3}
+        onClick={handleClick(setActive)}
+        icon={<IconHome aria-hidden />}
+      />
+      <SideNavigation.MainLevel
+        active={ID.MAIN_LEVEL_4 === active}
+        id={ID.MAIN_LEVEL_4}
+        href={ID.MAIN_LEVEL_4}
+        label={labels.mainLevel4}
+        external
+        onClick={handleClick(setActive)}
+        withDivider
+      />
+    </SideNavigation>
+  );
+};
+
+export const AutoCollapseOtherOpened = (args) => {
+  const [active, setActive] = React.useState(ID.SUB_LEVEL_1);
+
+  const handleMainLevelClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
+    ev.preventDefault();
+  };
+
+  const handleSubLevelClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
+    ev.preventDefault();
+    setActive((ev.currentTarget as HTMLAnchorElement).getAttribute('href') as ID);
+  };
+
+  return (
+    <SideNavigation {...args}>
+      <SideNavigation.MainLevel
+        active={ID.MAIN_LEVEL_1 === active}
+        id={ID.MAIN_LEVEL_1}
+        label={labels.mainLevel1}
+        onClick={handleMainLevelClick}
+      >
+        <SideNavigation.SubLevel
+          active={ID.SUB_LEVEL_1 === active}
+          id={ID.SUB_LEVEL_1}
+          href={ID.SUB_LEVEL_1}
+          label={labels.subLevel1}
+          onClick={handleSubLevelClick}
+        />
+        <SideNavigation.SubLevel
+          active={ID.SUB_LEVEL_2 === active}
+          id={ID.SUB_LEVEL_2}
+          href={ID.SUB_LEVEL_2}
+          label={labels.subLevel2}
+          onClick={handleSubLevelClick}
+        />
+      </SideNavigation.MainLevel>
+      <SideNavigation.MainLevel
+        active={ID.MAIN_LEVEL_2 === active}
+        id={ID.MAIN_LEVEL_2}
+        label={labels.mainLevel2}
+        onClick={handleMainLevelClick}
+      >
+        <SideNavigation.SubLevel
+          active={ID.SUB_LEVEL_3 === active}
+          id={ID.SUB_LEVEL_3}
+          href={ID.SUB_LEVEL_3}
+          label={labels.subLevel3}
+          onClick={handleSubLevelClick}
+        />
+        <SideNavigation.SubLevel
+          active={ID.SUB_LEVEL_4 === active}
+          id={ID.SUB_LEVEL_4}
+          href={ID.SUB_LEVEL_4}
+          label={labels.subLevel4}
+          onClick={handleSubLevelClick}
+        />
+      </SideNavigation.MainLevel>
+      <SideNavigation.MainLevel
+        active={ID.MAIN_LEVEL_3 === active}
+        id={ID.MAIN_LEVEL_3}
+        href={ID.MAIN_LEVEL_3}
+        label={labels.mainLevel3}
+        onClick={handleSubLevelClick}
+      />
+    </SideNavigation>
+  );
+};
+AutoCollapseOtherOpened.args = { autoCollapseOthers: true };
+
+export const CustomTheme = (args) => Default(args);
+CustomTheme.args = {
```

Comment:
I think the custom theme Storybook example has some contrast issues that we probably do not want to display as an example. I can alter the example so it is more in line with the Hel.fi color palette approach.